libs/langchain-community/src/document_loaders/web/airtable.ts

```diff
@@ -0,0 +1,221 @@
+   *
+   * @param tableId - ID of the Airtable table.
+   * @param baseId - ID of the Airtable base.
+   * @param kwargs - Additional query parameters for Airtable requests.
+   * @param config - Loader configuration for retry options.
+   */
+  constructor({ tableId, baseId, kwargs = {} }: AirtableLoaderOptions) {
+    super();
+    this.apiToken = getEnvironmentVariable("AIRTABLE_API_TOKEN") || "";
+    this.tableId = tableId;
+    this.baseId = baseId;
+    this.kwargs = kwargs;
+
+    if (!this.apiToken) {
+      throw new Error(
+        "Missing Airtable API token. Please set AIRTABLE_API_TOKEN environment variable."
+      );
+    }
+  }
+
+  /**
+   * Loads documents from Airtable, handling pagination and retries.
+   *
+   * @returns A promise that resolves to an array of Document objects.
+   */
+  public async load(): Promise<Document[]> {
+    const documents: Document[] = [];
+    let offset: string | undefined;
+
+    try {
+      do {
+        const url = this.constructUrl(offset);
+        const data = await this.retryFetchRecords(url);
+        data.records.forEach((record: AirtableRecord) =>
+          documents.push(this.createDocument(record))
+        );
+        offset = data.offset;
+      } while (offset);
+    } catch (error) {
+      console.error("Error loading Airtable records:", error);
+      throw new Error("Failed to load Airtable records");
+    }
+
+    return documents;
+  }
+
+  /**
+   * Asynchronous generator function for lazily loading documents from Airtable.
+   * This method yields each document individually, enabling memory-efficient
+   * handling of large datasets by fetching records in pages.
+   *
+   * @returns An asynchronous generator yielding Document objects one by one.
+   */
+  public async *loadLazy(): AsyncGenerator<Document> {
+    let offset: string | undefined;
+    try {
+      do {
+        const url = this.constructUrl(offset);
+        const data = await this.retryFetchRecords(url);
+
+        for (const record of data.records) {
+          yield this.createDocument(record);
+        }
+
+        offset = data.offset;
+      } while (offset);
+    } catch (error) {
+      console.error("Error loading Airtable records lazily:", error);
+      throw new Error("Failed to load Airtable records lazily");
+    }
+  }
+
+  /**
+   * Fetches records from Airtable with retry logic to handle transient failures.
+   *
+   * @param url - The Airtable API request URL.
+   * @param attempt - Current retry attempt number.
+   * @returns A promise that resolves to an AirtableResponse object.
+   */
+  private async retryFetchRecords(
+    url: string,
+    attempt = 1
+  ): Promise<AirtableResponse> {
+    try {
+      return await this.fetchRecords(url);
+    } catch (error) {
+      const maxRetries = AirtableLoader.MAX_RETRIES;
+      const retryDelayMs = AirtableLoader.DEFAULT_RETRY_DELAY_MS;
+
+      if (attempt <= maxRetries) {
```

Comment:
Generally prefer the builtin `AsyncCaller` for consistency:

https://github.com/langchain-ai/langchainjs/blob/main/langchain-core/src/utils/async_caller.ts

How to use (it's a built-in field on the base `Embeddings` class): https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-openai/src/chat_models.ts#L1821

Reply:
Import example

https://github.com/langchain-ai/langchainjs/blob/e54c10124d793614031664367622271da8e35114/langchain/src/util/prompt-layer.ts#L2